.eslintrc

```diff
@@ -1,25 +1,60 @@
 {
-	"extends": [
-		"standard",
-		"plugin:jest/recommended",
-		"plugin:import/recommended",
-		"plugin:n/recommended",
-		"plugin:prettier/recommended",
-		"plugin:promise/recommended"
-	],
+	"parser": "@typescript-eslint/parser",
 	"parserOptions": {
-		"ecmaVersion": 2018
-	},
-	"globals": {
-		"define": "readonly"
+		"project": "tsconfig.json",
+		"sourceType": "module"
 	},
+	"plugins": ["@typescript-eslint/eslint-plugin"],
+	"extends": [
+		"plugin:@typescript-eslint/recommended",
+		"plugin:@typescript-eslint/recommended-requiring-type-checking",
+		"plugin:@typescript-eslint/strict",
+		"plugin:prettier/recommended"
+	],
+	"root": true,
 	"env": {
-		"browser": true,
-		"es6": true,
-		"node": true,
-		"jest/globals": true
+		"node": true
```

Comment:
should we maybe use `shared-node-browser` here since this library should be usable in both? https://eslint.org/docs/latest/use/configure/language-options#specifying-environments

Reply:
that's a good point for when we'll be adding browser support, but for now, it cannot run there so I'm only going to link that discussion in the relevant issue/PR if that's ok for you.